docs/docs/adding-search-with-js-search.md

```diff
@@ -1,8 +1,357 @@
 ---
-title: Adding search with js-search
+title: "Adding search with js-search"
 ---
 
-This is a stub. Help our community expand it.
+## Prerequisites
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Before we go through the steps needed for adding client side search to your Gatsby website, it would be advised
+if the reader is not familiar on how Gatsby works and is set up to follow through the [tutorial](https://www.gatsbyjs.org/tutorial/) and brush up on the [documentation](https://www.gatsbyjs.org/docs/).
+
+Otherwise just skip this part and move onto the next part.
+
+## What is JS Search
+
+[JS Search](https://github.com/bvaughn/js-search) is a library created by Brian Vaughn, a member of the core team at Facebook. It provides an efficient way to search for data on the client using JavaScript and JSON objects. It also has extensive customisation options, check out their docs for more details.
+
+The full code and documentation for this library is [available on GitHub](https://github.com/bvaughn/js-search). This guide is based on the official js-search example but has been adapted to work with your Gatsby site.
+
+## Setup
+
+Let's start by creating a new Gatsby site to work with. Open up a terminal and create the folder that you'll use for this project:
+
+```bash
+mkdir client-search-with-gatsby
+```
+
+Inside that folder, create a new Gatsby website using a starter template, using the command below:
+
+```bash
+npx gatsby new jsSearchExample https://github.com/gatsbyjs/gatsby-starter-default
```

Comment:
Why not propose a compromise, instead of all lowercase how about `js-search-example`, reason being because someone, let's call it the better half, while taking a peak at the document, she said and i quote, "Change the folder name to a hyfenated name, because and you know me i'm not a tech expert, if i was to pickup on this i would like the folder name to be as humanly readable as possible, even not being a native english speaking person" and relationship aside, she's got a point in there. 😉

Reply:
@jonniebigodes that sounds great! I think we should generally use dash-case (often called kebab-case), so yeah, good call!